### PR TITLE
Update statistics.jl

### PR DESCRIPTION
**Pull request by HanLi123**
add method to get  Pearson correlation and covariance of two vectors.

### CODE REVIEW

**Comment by GiggleLiu** [comment]
Looks good, do you might adding a test here: https://github.com/GiggleLiu/NiLang.jl/blob/master/test/stdlib/statistics.jl

**Comment by HanLi123** [comment]
@GiggleLiu  Done.

File: src/stdlib/statistics.jl [hunk 1]
```diff
@@ -48,3 +48,38 @@ get the pdf of `Normal(μ, σ)` at point `x`.
...
+@i function i_cor_cov(rho!::T,cov!::T,a::AbstractVector{T},b::AbstractVector{T}) where T
+    @routine  begin
+        @zeros T var1 varsum1 mean1 sum1 std1
+        i_var_mean_sum(var1, varsum1, mean1, sum1, a)
+        std1 += sqrt(var1)
+        @zeros T var2 varsum2 mean2 sum2 std2
+        i_var_mean_sum(var2, varsum2, mean2, sum2, b)
+        std2 += sqrt(var2)
+        @zeros T anc3 anc4 anc5 anc6 anc7
+        for i=1:length(b)
```
Comment (GiggleLiu) [inline src/stdlib/statistics.jl:68]:
```suggestion
        @inbounds for i=1:length(b)
```

File: src/stdlib/statistics.jl [hunk 2]
```diff
@@ -48,3 +48,38 @@ get the pdf of `Normal(μ, σ)` at point `x`.
...
+
+"""
+     i_cor_cov(rho!,cov!,a,b)
+
+     get Pearson correlation and covariance of two vectors a and b 
+
+"""
+
+@i function i_cor_cov(rho!::T,cov!::T,a::AbstractVector{T},b::AbstractVector{T}) where T
+    @routine  begin
```
Comment (GiggleLiu) [inline src/stdlib/statistics.jl:60]:
```suggestion
    @routine  @invcheckoff begin
```

File: src/stdlib/statistics.jl [hunk 3]
```diff
@@ -48,3 +48,38 @@ get the pdf of `Normal(μ, σ)` at point `x`.
...
 end
+
+"""
+     i_cor_cov(rho!,cov!,a,b)
+
+     get Pearson correlation and covariance of two vectors a and b 
+
+"""
+
+@i function i_cor_cov(rho!::T,cov!::T,a::AbstractVector{T},b::AbstractVector{T}) where T
```
Comment (GiggleLiu) [inline src/stdlib/statistics.jl:59]:
```suggestion
@i function i_cor_cov(rho!::T,cov!::T,a::AbstractVector{T},b::AbstractVector{T}) where T
    @safe @assert length(a) == length(b)
```

File: src/stdlib/statistics.jl [hunk 4]
```diff
@@ -48,3 +48,38 @@ get the pdf of `Normal(μ, σ)` at point `x`.
 
     ~@routine
 end
+
+"""
+     i_cor_cov(rho!,cov!,a,b)
+
+     get Pearson correlation and covariance of two vectors a and b
```
Comment (GiggleLiu) [inline src/stdlib/statistics.jl:55]:
```suggestion
get Pearson correlation and covariance of two vectors `a` and `b` 
```

File: src/stdlib/statistics.jl [hunk 5]
```diff
@@ -1,4 +1,4 @@
-export i_mean_sum, i_var_mean_sum, i_normal_logpdf
+export i_mean_sum, i_var_mean_sum, i_normal_logpdf,i_cor_cov
```
Comment (GiggleLiu) [inline src/stdlib/statistics.jl:1]:
```suggestion
export i_mean_sum, i_var_mean_sum, i_normal_logpdf, i_cor_cov
```

**Review by GiggleLiu — APPROVED** [review]

**Comment by GiggleLiu** [comment]
The code looks good in general, I just nitpicked some details. e.g. adding `@inbounds`.

Thank you so much for contributing NiLang!

**Comment by GiggleLiu** [comment]
I polished the code a bit, hopefully these changes would make the code run faster.
See:
https://github.com/GiggleLiu/NiLang.jl/pull/33/files

**Pull request merged**